src/microbe_stage/organelle_unlocks/StatisticBasedUnlockCondition.cs

```diff
@@ -0,0 +1,181 @@
+        public int Required { get; set; }
+
+        public override bool Satisfied()
+        {
+            if (RelevantStatistic == null)
+                return false;
+
+            return RelevantStatistic.Value >= Required;
+        }
+
+        public override void GenerateTooltip(LocalizedStringBuilder builder)
+        {
+            if (RelevantStatistic == null)
+                return;
+
+            builder.Append(new LocalizedString("UNLOCK_CONDITION_ENGULFED_MICROBES_ABOVE", Required, RelevantStatistic.Value));
+        }
+    }
+
+    /// <summary>
+    ///   The number of times the player has died is above a specified amount.
+    /// </summary>
+    public class PlayerDeathsAbove : StatisticBasedUnlockCondition<SimpleStatistic>
+    {
+        [JsonIgnore]
+        public override StatsTrackerEvent RelevantEvent => StatsTrackerEvent.PlayerDied;
+
+        [JsonProperty]
+        public int Required { get; set; }
+
+        public override bool Satisfied()
+        {
+            if (RelevantStatistic == null)
+                return false;
+
+            return RelevantStatistic.Value >= Required;
+        }
+
+        public override void GenerateTooltip(LocalizedStringBuilder builder)
+        {
+            if (RelevantStatistic == null)
+                return;
+
+            builder.Append(new LocalizedString("UNLOCK_CONDITION_PLAYER_DEATH_COUNT_ABOVE", Required, RelevantStatistic.Value));
+        }
+    }
+
+    public class ReproduceInBiome : StatisticBasedUnlockCondition<ReproductionStatistic>
+    {
+        [JsonIgnore]
+        public override StatsTrackerEvent RelevantEvent => StatsTrackerEvent.PlayerReproduced;
+
+        [JsonProperty]
+        public Biome? Biome { get; set; }
+
+        public override void GenerateTooltip(LocalizedStringBuilder builder)
+        {
+            if (RelevantStatistic == null)
+                return;
+
+            builder.Append(new LocalizedString("UNLOCK_CONDITION_REPRODUCE_IN_BIOME", Biome!.Name));
+        }
+
+        public override bool Satisfied()
+        {
+            if (RelevantStatistic == null)
+                return false;
+
+            if (!RelevantStatistic.ReproducedInBiomes.TryGetValue(Biome!, out var count))
+                return false;
+
+            return count >= 1;
+        }
+
+        public override void Check(string name)
+        {
+            if (Biome == null)
+            {
+                throw new InvalidRegistryDataException(name, GetType().Name,
+                    "Biome is null");
+            }
+        }
+    }
+
+    public class ReproduceWithOrganelle : StatisticBasedUnlockCondition<ReproductionStatistic>
+    {
+        [JsonIgnore]
+        private OrganelleDefinition organelle = null!;
+
+        [JsonIgnore]
+        public override StatsTrackerEvent RelevantEvent => StatsTrackerEvent.PlayerReproduced;
+
+        [JsonProperty("Organelle")]
+        public string RawOrganelle { get; set; } = string.Empty;
```

Comment:
There's still this raw value business here.

Reply:
Wouldn't this cause some type of circular dependency? Unlock conditions are defined in organelle definitions, and in this case the unlock conditions would be referencing those organelle definitions. Wouldn't that mean that this could reference not yet defined organelles?

Reply:
Yeah that would cause a circular dependency. So this will likely have to stay. Would be nice to add a documentation comment here explaining that reasoning.